src/SourcePathsMapping/include/SourcePathsMapping/MappingManager.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SOURCE_PATHS_MAPPING_MAPPING_MANAGER_H_
+#define SOURCE_PATHS_MAPPING_MAPPING_MANAGER_H_
+
+#include <optional>
+#include <vector>
+
+#include "SourcePathsMapping/Mapping.h"
+
+namespace orbit_source_paths_mapping {
+
+// Maintains a list of source paths mappings and takes care of persistence using QSettings.
+class MappingManager {
+ public:
+  explicit MappingManager();
+  [[nodiscard]] const std::vector<Mapping>& GetMappings() const { return mappings_; }
+
+  void SetMappings(std::vector<Mapping> mappings);
+  void AppendMapping(Mapping mapping);
+
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstMatchingTarget(
+      const std::filesystem::path& source_path) const {
+    return orbit_source_paths_mapping::MapToFirstMatchingTarget(mappings_, source_path);
+  }
+
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstExistingTarget(
```

Comment:
Some comment about the difference between these 2 functions? I spent a little there trying to understand.